### PR TITLE
[Relay][BYOCG] Propagate constant to subgraphs

### PR DESCRIPTION
**Pull request by zhiics**
Currently, all constants are not propagated to the functions that will be handled by external codegen. This may cause some performance issue if a function is frequently used as you will need to pass the constants multiple times to accelerates. This PR binds the constants to the new functions and propagate them. As expected, this will need the 3rd-party codegen to handle the constants by themselves.

@comaniac @mbaret @soiferj @trevor-m @tqchen

### CODE REVIEW

**Comment by trevor-m** [comment]
Thank you!
I have tested this for my external codegen and it fixes a few problems I was having.

**Comment by mbaret** [comment]
I think @lhutton1 has had an issue that might be related, maybe he can take a look.

**Comment by lhutton1** [comment]
Just tested it out with my codegen and works well, thanks for this!

File: src/relay/backend/contrib/codegen_c/codegen.cc [hunk 1]
```diff
@@ -40,14 +41,63 @@ class CodegenC : public ExprVisitor, public CodegenCBase {
...
+  void VisitExpr_(const VarNode* node) final {
     ext_func_args_.push_back(GetRef<Var>(node));
     out_.clear();
     Output output;
     output.name = node->name_hint();
     out_.push_back(output);
   }
 
+  void VisitExpr_(const ConstantNode* cn) final {
+    Constant con = GetRef<Constant>(cn);
```
Comment (anijain2305) [inline src/relay/backend/contrib/codegen_c/codegen.cc]: s/con/constant

Very minor, feel free to ignore.

**Comment by zhiics** [comment]
Thanks everybody. This is now merged.

**Pull request merged**